testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksChangeDataFeedCompatibility.java

```diff
@@ -0,0 +1,219 @@
+
+            assertThat(onDelta().executeQuery(
+                    "SELECT updated_column, partitioning_column_1, partitioning_column_2, _change_type, _commit_version " +
+                            "FROM table_changes('default." + tableName + "', 0)"))
+                    .containsOnly(
+                            row("testValue1", 1, "partition1", "insert", 1L),
+                            row("testValue2", 2, "partition2", "insert", 1L),
+                            row("testValue3", 3, "partition3", "insert", 1L),
+                            row("testValue1", 1, "partition1", "update_preimage", 2L),
+                            row("testValue1", 5, "partition1", "update_postimage", 2L),
+                            row("testValue2", 2, "partition2", "update_preimage", 3L),
+                            row("testValue4", 4, "partition2", "update_postimage", 3L));
+        }
+        finally {
+            onDelta().executeQuery("DROP TABLE IF EXISTS default." + tableName);
+        }
+    }
+
+    @Test(groups = {DELTA_LAKE_DATABRICKS, DELTA_LAKE_EXCLUDE_73, PROFILE_SPECIFIC_TESTS})
+    public void testUpdateTableWithCDFEnabledAfterTableIsAlreadyCreated()
+    {
+        String tableName = "test_updates_to_table_with_cdf_enabled_later_" + randomNameSuffix();
+        try {
+            onDelta().executeQuery("CREATE TABLE default." + tableName + " (col1 STRING, updated_column INT) " +
+                    "USING DELTA " +
+                    "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'");
+
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES('testValue1', 1)");
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES('testValue2', 2)");
+            onDelta().executeQuery("INSERT INTO default." + tableName + " VALUES('testValue3', 3)");
+            onTrino().executeQuery("UPDATE delta.default." + tableName + " SET updated_column = 5 WHERE col1 = 'testValue3'");
+            onDelta().executeQuery("ALTER TABLE default." + tableName + " SET TBLPROPERTIES (delta.enableChangeDataFeed = true)");
+            onTrino().executeQuery("UPDATE delta.default." + tableName + " SET updated_column = 4 WHERE col1 = 'testValue3'");
+
+            assertQueryFailure(() -> onDelta().executeQuery("SELECT col1, updated_column, _change_type, _commit_version FROM table_changes('default." + tableName + "', 0)"))
+                    .hasMessageContaining("Error getting change data for range [0 , 6] as change data was not");
+
+            onTrino().executeQuery("INSERT INTO delta.default." + tableName + " VALUES('testValue6', 6)");
+            assertThat(onDelta().executeQuery("SELECT col1, updated_column, _change_type, _commit_version FROM table_changes('default." + tableName + "', 5)"))
```

Comment:
Instead of hardcoding table versions in the test, let's use a utility method based on `DESCRIBE HISTORY tblName` command from Databricks to obtain the latest table version. This would eventually make the test more robust.

Reply:
it would but here I need not the latest version but latest - 2, do you think it would still be more readable ?

Reply:
You can save the actual version (a few lines above) from which you want to get the table changes in a variable and you can use it then in `table_changes` call.